test/foundry/OffchainAssetReceiptVaultFactory.t.sol

```diff
@@ -0,0 +1,68 @@
+pragma solidity =0.8.17;
+
+import "../../contracts/vault/receipt/ReceiptVault.sol";
+import "../../contracts/vault/receipt/ReceiptFactory.sol";
+import "../../contracts/test/CreateOffchainAssetReceiptVaultFactory.sol";
+
+import "forge-std/Test.sol";
+
+import "../../contracts/vault/offchainAsset/OffchainAssetReceiptVaultFactory.sol";
+import "../../contracts/vault/offchainAsset/OffchainAssetReceiptVault.sol";
+
+contract OffChainAssetReceiptVaultFactoryTest is Test, CreateOffchainAssetReceiptVaultFactory {
+    OffchainAssetVaultConfig offchainAssetVaultConfig;
+    VaultConfig vaultConfig;
+    OffchainAssetReceiptVault vault;
+
+    function setUp() public {
+        createOffchainAssetReceiptVaultFactory();
+    }
+
+    function testOffchainAssetReceiptVaultFactoryConstuction() public view {
+        assert(address(factory) != address(0));
+    }
+
+    function testCreateChild(address alice, string memory assetName, string memory assetSymbol) public {
+        // Exclude the zero address
+        vm.assume(alice != address(0));
```

Comment:
this doesn't actually avoid all dangerous alice values, which can include the vm itself and precompiles, probably better to generate addresses using the fuzzed input as seed